File: src/r_2_pattern/Main.java
/**
 * @author Pranshu Aggarwal
 * @problem https://hack.codingblocks.com/app/practice/1/54/problem
 */
package r_2_pattern;

import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Main {
	static void r2_pattern_recursive(int n) {
		if(n<1) return;
		if(n % 2 != 0) System.out.println(n);
		r2_pattern_recursive(n-1);
		if(n % 2 == 0 && n != 0) System.out.println(n);
	}
	static void r2_pattern_iterative(int n) {
		for(int i = n; i > 0; i--) if(i % 2 != 0) System.out.println(i);
		for(int i = 2; i <= n; i++) if(i % 2 == 0) System.out.println(i);
	}
	public static void main(String[] args) throws Exception {
		BufferedReader sc = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(sc.readLine().trim());
		r2_pattern_iterative(n);
//		r2_pattern_recursive(n);
	}

}

File: src/r_2_pattern/main.py
'''
 @author: Pranshu Aggarwal
 @problem: https://hack.codingblocks.com/app/practice/1/54/problem
'''
def r2_pattern_recursive(n):
    if n < 1: return
    if n % 2 is not 0: print(n)
    r2_pattern_recursive(n-1)
    if n % 2 is 0: print(n)

def r2_pattern_iterative(n):
    for i in range(n, 0, -1):
        if i % 2 != 0: print(i)
    for i in range(2, n+1, 1):
        if i % 2 == 0: print(i)

if __name__ == "__main__":
    n = int(input().strip())
    r2_pattern_iterative(n)
#     r2_pattern_recursive(n)